ui-v2/tests/unit/utils/dom/event-source/cache-test.js

```diff
@@ -0,0 +1,136 @@
+import domEventSourceCache from 'consul-ui/utils/dom/event-source/cache';
+import { module } from 'qunit';
+import test from 'ember-sinon-qunit/test-support/test';
+
+module('Unit | Utility | dom/event-source/cache');
+
+const createEventSource = function() {
+  return class {
+    constructor(cb) {
+      this.source = cb;
+      this.source.apply(this, arguments);
+    }
+    addEventListener() {}
+    removeEventListener() {}
+    dispatchEvent() {}
+    close() {}
+  };
+};
```

Comment:
I think I was probably trying to keep things consistent. The user facing API uses a factory function, and I'm also using a factory function to create my mock promises.

It also means that if at some point I do multiple tests in the same file using the mock EventSource class, I get a fresh instance of the actual class every time (as opposed to an instance *based* on the class). It's not important right now I don't think, as there's nothing there that would produce a bug due to using the same class instance across tests. But each test is supposed to be isolated, and theoretically this means they are. This reason is probably accidental, I would imagine I was just trying to be consistent, or trying to plan ahead in case I needed to mock and inject anything further to create the class.

Reply:
Consistency is fine, but isolation is an incorrect reason. It's a static class and you are only operating on instances of the class. Wrapping the whole thing in a function doesn't give you any further isolation, it just redundantly creates class definitions. If the VM is smart enough, it probably just JITs that wrapping function away anyway.